src/Akka.Persistence.Redis.Hosting/AkkaPersistenceMongoDbHostingExtensions.cs

```diff
@@ -0,0 +1,216 @@
+using System;
+using System.Text;
+using Akka.Actor;
+using Akka.Configuration;
+using Akka.Hosting;
+using Akka.Persistence.Hosting;
+
+#nullable enable
+namespace Akka.Persistence.Redis.Hosting;
+
+public static class AkkaPersistenceRedisHostingExtensions
+{
+    /// <summary>
+    ///     Adds Akka.Persistence.Redis support to this <see cref="ActorSystem"/>.
+    /// </summary>
+    /// <param name="builder">
+    ///     The builder instance being configured.
+    /// </param>
+    /// <param name="configurationString">
+    ///     Connection string as described here: https://stackexchange.github.io/StackExchange.Redis/Configuration#basic-configuration-strings.
+    /// </param>
+    /// <param name="mode">
+    ///     <para>
+    ///         Determines which settings should be added by this method call.
+    ///     </para>
+    ///     <i>Default</i>: <see cref="PersistenceMode.Both"/>
+    /// </param>
+    /// <param name="autoInitialize">
+    ///     <para>
+    ///         Should the redis store table be initialized automatically.
+    ///     </para>
+    ///     <i>Default</i>: <c>false</c>
+    /// </param>
+    /// <param name="journalBuilder">
+    ///     <para>
+    ///         An <see cref="Action"/> used to configure an <see cref="AkkaPersistenceJournalBuilder"/> instance.
+    ///     </para>
+    ///     <i>Default</i>: <c>null</c>
+    /// </param>
+    /// <param name="pluginIdentifier">
+    ///     <para>
+    ///         The configuration identifier for the plugins
+    ///     </para>
+    ///     <i>Default</i>: <c>"redis"</c>
+    /// </param>
+    /// <param name="isDefaultPlugin">
+    ///     <para>
+    ///         A <c>bool</c> flag to set the plugin as the default persistence plugin for the <see cref="ActorSystem"/>
+    ///     </para>
+    ///     <b>Default</b>: <c>true</c>
+    /// </param>
+    /// <returns>
+    ///     The same <see cref="AkkaConfigurationBuilder"/> instance originally passed in.
+    /// </returns>
+    /// <exception cref="ArgumentOutOfRangeException">
+    ///     Thrown when <see cref="journalBuilder"/> is set and <see cref="mode"/> is set to
+    ///     <see cref="PersistenceMode.SnapshotStore"/>
+    /// </exception>
+    public static AkkaConfigurationBuilder WithRedisPersistence(
+        this AkkaConfigurationBuilder builder,
+        string configurationString,
```

Comment:
Through out the solution there is an inconsistency where the setting is called configuration string but all the documentation refers to it as a connection string. The changes matches what has originally been in place. This can be changed to normalise on connection string if needed

https://github.com/akkadotnet/Akka.Persistence.Redis/blob/389d3e2d28a129b803fec7447085f519342a6ba3/src/Akka.Persistence.Redis/reference.conf#L7

Reply:
I can see why this can be a confusion point, Redis themselves called it "Configuration" (https://stackexchange.github.io/StackExchange.Redis/Configuration.html)